frontend/src/components/post/CreatePostModal.jsx

```diff
@@ -66,6 +67,52 @@ class CreatePostModal extends React.Component {
         // Clear all the content of the modal as well
     }
 
+    getBase64(e) {
+        var file = e.target.files[0];
+        if (file === undefined) {
+          return;
+        } else {
+          let reader = new FileReader();
+          reader.readAsDataURL(file);
+          reader.onload = () => {
+            this.setState(({image:reader.result}))
+          };
+          reader.onerror = (err) => {
+            console.log(err);
```

Comment:
One thing, i think you should set the msg field if there is a reader error and let the user know something went wrong on uplaod